src/main/java/org/folio/rest/impl/ProxiesForAPI.java

```diff
@@ -40,320 +36,116 @@
-                    } else {
-                      asyncResultHandler.handle(Future.succeededFuture(
-                              PostProxiesforResponse.respond500WithTextPlain(
-                                      getErrorResponse(message))));
-                    }
-                  } else {
-                    String returnObject = reply.result();
-                    entity.setId(returnObject);
-                    asyncResultHandler.handle(Future.succeededFuture(
-                      PostProxiesforResponse.respond201WithApplicationJson(entity,
-                        PostProxiesforResponse.headersFor201().withLocation(URL_PREFIX + returnObject))));
-                  }
-                } catch(Exception e) {
-                  String message = logAndSaveError(e);
-                  asyncResultHandler.handle(Future.succeededFuture(
-                              PostProxiesforResponse.respond500WithTextPlain(
-                                      getErrorResponse(message))));
-                }
-              });
-            } catch(Exception e) {
-              String message = logAndSaveError(e);
-              asyncResultHandler.handle(Future.succeededFuture(
-                      PostProxiesforResponse.respond500WithTextPlain(
-                              getErrorResponse(message))));
-            }
+    String lang,
+    ProxiesFor entity,
+    Map<String, String> okapiHeaders,
+    Handler<AsyncResult<Response>> asyncResultHandler,
+    Context vertxContext) {
+    PostgresClient postgresClient = PostgresClientUtil.getInstance(vertxContext, okapiHeaders);
+    userAndProxyUserComboExists(entity.getUserId(), entity.getProxyUserId(),
+      postgresClient).setHandler(existsRes -> {
+        if (existsRes.failed()) {
+          String message = logAndSaveError(existsRes.cause());
+          asyncResultHandler.handle(Future.succeededFuture(
+            PostProxiesforResponse.respond500WithTextPlain(
+              getErrorResponse(message))));
+        } else if (existsRes.result() == true) {
+          Errors existsError = ValidationHelper.createValidationErrorMessage(
+            "proxyFor", entity.getId(), "Proxy relationship already exists");
+          asyncResultHandler.handle(Future.succeededFuture(
+            PostProxiesforResponse.respond422WithApplicationJson(existsError)));
+        } else {
+          String id = entity.getId();
+          if (id == null) {
+            id = UUID.randomUUID().toString();
+            entity.setId(id);
           }
-        });
-      } catch(Exception e) {
-        String message = logAndSaveError(e);
-        asyncResultHandler.handle(Future.succeededFuture(
-                    PostProxiesforResponse.respond500WithTextPlain(
-                            getErrorResponse(message))));
-      }
-    });
+          PgUtil.post(PROXY_FOR_TABLE, entity, okapiHeaders, vertxContext,
+            PostProxiesforResponse.class, asyncResultHandler);
+        }
+      });
```

Comment:
Improve readability by using guard clauses. This will reduce the indentation level of the PgUtil.post, the main path of the execution flow:

    if (existsRes.failed()) {
      …
      return;
    }
    if (existsRes.result() == true) {
      …
      return;
    }
    PgUtil.post(…

Reply:
sure